android/java/org/chromium/chrome/browser/crypto_wallet/activities/NftDetailActivity.java

```diff
@@ -0,0 +1,172 @@
+            PortfolioModel.Erc721MetaData erc721MetaData =
+                    (PortfolioModel.Erc721MetaData) getIntent().getSerializableExtra(
+                            Utils.NFT_META_DATA);
+            setMetadata(erc721MetaData);
+        }
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        Button btnSend = findViewById(R.id.btn_send);
+        // TODO(simone): Enable if it's the NFT owner.
+        btnSend.setVisibility(View.GONE);
+
+        TextView nftDetailTitle = findViewById(R.id.nft_detail_title);
+        nftDetailTitle.setText(Utils.formatErc721TokenTitle(mNftName, mNftTokenId));
+
+        TextView nftName = findViewById(R.id.nft_name);
+        nftName.setText(mNftName);
+
+        TextView networkName = findViewById(R.id.blockchain_content);
+        networkName.setText(mNetworkName);
+        TextView tokenStandard = findViewById(R.id.token_standard_content);
+        tokenStandard.setText(R.string.brave_wallet_nft_erc_721);
+        TextView tokenIdView = findViewById(R.id.token_id_content);
+
+        String tokenStr = String.format(TOKEN_ID_FORMAT, mNftTokenId);
+        SpannableString spannable = new SpannableString(tokenStr);
+
+        if (!TextUtils.isEmpty(mBlockExplorerUrl)) {
+            String url =
+                    String.format(NFT_URL_FORMAT, mBlockExplorerUrl, mContractAddress, mNftTokenId);
+            NoUnderlineClickableSpan linkSpan = new NoUnderlineClickableSpan(this,
+                    R.color.brave_link, (textView) -> { TabUtils.openLinkWithFocus(this, url); });
+            spannable.setSpan(linkSpan, 0, spannable.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        }
+        tokenIdView.setText(spannable);
+        tokenIdView.setMovementMethod(LinkMovementMethod.getInstance());
+        onInitialLayoutInflationComplete();
+    }
+
+    private void setMetadata(PortfolioModel.Erc721MetaData erc721MetaData) {
+        // In case of no errors proceed to assign description and fetch NFT image.
+        if (erc721MetaData.mErrCode == 0) {
+            String description = erc721MetaData.mDescription;
+            String imageUrl = erc721MetaData.mImageUrl;
+            if (!TextUtils.isEmpty(description)) {
+                TextView descriptionContent = findViewById(R.id.description_content);
+                descriptionContent.setText(description);
+            } else {
+                // Hide description layout in case of an empty string.
+                findViewById(R.id.nft_description).setVisibility(View.GONE);
+            }
+            if (!TextUtils.isEmpty(imageUrl)) {
+                loadNftImage(imageUrl);
+            } else {
+                setNftImageAsNotAvailable();
+            }
+        }
+    }
+
+    private void loadNftImage(String imageUrl) {
+        ImageLoader.getLoadNftRequest(imageUrl, this, false)
```

Comment:
trivial suggestion: `makeLoadNftRequest` or `createLoadNftRequest` would better explain the purpose.